<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>930</width>
    <height>970</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Feistel Function</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="3">
     <widget class="QSpinBox" name="spinMsg0">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>64</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>64</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="minimum">
       <number>0</number>
      </property>
      <property name="maximum">
       <number>255</number>
      </property>
      <property name="value">
       <number>8</number>
      </property>
      <property name="displayIntegerBase">
       <number>16</number>
      </property>
     </widget>
    </item>
    <item row="0" column="6">
     <widget class="QSpinBox" name="spinMsg3">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>64</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>64</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="minimum">
       <number>0</number>
      </property>
      <property name="maximum">
       <number>255</number>
      </property>
      <property name="value">
       <number>8</number>
      </property>
      <property name="displayIntegerBase">
       <number>16</number>
      </property>
     </widget>
    </item>
    <item row="4" column="10">
     <widget class="QLabel" name="iconCredit">
      <property name="font">
       <font>
        <pointsize>6</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Freepik</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
    </item>
    <item row="0" column="4">
     <widget class="QSpinBox" name="spinMsg1">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>64</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>64</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="minimum">
       <number>0</number>
      </property>
      <property name="maximum">
       <number>255</number>
      </property>
      <property name="value">
       <number>8</number>
      </property>
      <property name="displayIntegerBase">
       <number>16</number>
      </property>
     </widget>
    </item>
    <item row="5" column="1" colspan="10">
     <widget class="QQuickWidget" name="quickWidget">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>660</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="autoFillBackground">
       <bool>false</bool>
      </property>
      <property name="resizeMode">
       <enum>QQuickWidget::SizeRootObjectToView</enum>
      </property>
      <property name="source">
       <url>
        <string>qrc:/qml/feistel.qml</string>
       </url>
      </property>
     </widget>
    </item>
    <item row="2" column="2">
     <widget class="QLabel" name="labEnc">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>220</width>
        <height>0</height>
       </size>
      </property>
      <property name="text">
       <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Feistel function result &lt;span style=&quot; font-weight:600; font-style:italic;&quot;&gt;f &lt;/span&gt;(32 bit):&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
     </widget>
    </item>
    <item row="1" column="9">
     <spacer name="horizontalSpacer">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>71</width>
        <height>20</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="2" column="4">
     <widget class="QLabel" name="valEnc1">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>64</width>
        <height>0</height>
       </size>
      </property>
      <property name="font">
       <font>
        <pointsize>12</pointsize>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>C2</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item row="1" column="2">
     <widget class="QLabel" name="labKey">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>220</width>
        <height>0</height>
       </size>
      </property>
      <property name="text">
       <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Round key &lt;span style=&quot; font-weight:600;&quot;&gt;k&lt;/span&gt;&lt;span style=&quot; font-weight:600; vertical-align:sub;&quot;&gt;i&lt;/span&gt; (48 bit):&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
     </widget>
    </item>
    <item row="1" column="6">
     <widget class="QSpinBox" name="spinKey3">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>64</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>64</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="minimum">
       <number>0</number>
      </property>
      <property name="maximum">
       <number>255</number>
      </property>
      <property name="value">
       <number>51</number>
      </property>
      <property name="displayIntegerBase">
       <number>16</number>
      </property>
     </widget>
    </item>
    <item row="3" column="1" rowspan="2" colspan="5">
     <widget class="QLabel" name="label">
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>42</height>
       </size>
      </property>
      <property name="font">
       <font>
        <pointsize>10</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Click on message bit to toggle            . . .                  or enter hex values in spin box</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignBottom|Qt::AlignLeading|Qt::AlignLeft</set>
      </property>
     </widget>
    </item>
    <item row="0" column="10" rowspan="4">
     <widget class="QLabel" name="icon">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>128</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>128</width>
        <height>128</height>
       </size>
      </property>
      <property name="text">
       <string/>
      </property>
      <property name="pixmap">
       <pixmap resource="feistel.qrc">:/icons/merry-christmas.png</pixmap>
      </property>
      <property name="scaledContents">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item row="0" column="5">
     <widget class="QSpinBox" name="spinMsg2">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>64</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>64</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="minimum">
       <number>0</number>
      </property>
      <property name="maximum">
       <number>255</number>
      </property>
      <property name="value">
       <number>8</number>
      </property>
      <property name="displayIntegerBase">
       <number>16</number>
      </property>
     </widget>
    </item>
    <item row="2" column="6">
     <widget class="QLabel" name="valEnc3">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>64</width>
        <height>0</height>
       </size>
      </property>
      <property name="font">
       <font>
        <pointsize>12</pointsize>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>C4</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item row="1" column="3">
     <widget class="QSpinBox" name="spinKey0">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>64</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>64</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="minimum">
       <number>0</number>
      </property>
      <property name="maximum">
       <number>255</number>
      </property>
      <property name="value">
       <number>254</number>
      </property>
      <property name="displayIntegerBase">
       <number>16</number>
      </property>
     </widget>
    </item>
    <item row="1" column="8">
     <widget class="QSpinBox" name="spinKey5">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>64</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>64</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="minimum">
       <number>0</number>
      </property>
      <property name="maximum">
       <number>255</number>
      </property>
      <property name="value">
       <number>16</number>
      </property>
      <property name="displayIntegerBase">
       <number>16</number>
      </property>
     </widget>
    </item>
    <item row="1" column="4">
     <widget class="QSpinBox" name="spinKey1">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>64</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>64</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="minimum">
       <number>0</number>
      </property>
      <property name="maximum">
       <number>255</number>
      </property>
      <property name="value">
       <number>170</number>
      </property>
      <property name="displayIntegerBase">
       <number>16</number>
      </property>
     </widget>
    </item>
    <item row="0" column="2">
     <widget class="QLabel" name="labMsg">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>220</width>
        <height>0</height>
       </size>
      </property>
      <property name="text">
       <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Round input right half &lt;span style=&quot; font-weight:600;&quot;&gt;R&lt;/span&gt;&lt;span style=&quot; font-weight:600; vertical-align:sub;&quot;&gt;i&lt;/span&gt; (32 bit):&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
     </widget>
    </item>
    <item row="0" column="0" rowspan="4" colspan="2">
     <widget class="QLabel" name="spacer">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="2" column="5">
     <widget class="QLabel" name="valEnc2">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>64</width>
        <height>0</height>
       </size>
      </property>
      <property name="font">
       <font>
        <pointsize>12</pointsize>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>C3</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item row="1" column="7">
     <widget class="QSpinBox" name="spinKey4">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>64</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>64</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="minimum">
       <number>0</number>
      </property>
      <property name="maximum">
       <number>255</number>
      </property>
      <property name="value">
       <number>34</number>
      </property>
      <property name="displayIntegerBase">
       <number>16</number>
      </property>
     </widget>
    </item>
    <item row="1" column="5">
     <widget class="QSpinBox" name="spinKey2">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>64</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>64</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="minimum">
       <number>0</number>
      </property>
      <property name="maximum">
       <number>255</number>
      </property>
      <property name="value">
       <number>85</number>
      </property>
      <property name="displayIntegerBase">
       <number>16</number>
      </property>
     </widget>
    </item>
    <item row="2" column="3">
     <widget class="QLabel" name="valEnc0">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>64</width>
        <height>0</height>
       </size>
      </property>
      <property name="font">
       <font>
        <pointsize>12</pointsize>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>C1</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>930</width>
     <height>30</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuStyle">
    <property name="title">
     <string>Style</string>
    </property>
    <addaction name="actionDark"/>
    <addaction name="actionLight"/>
   </widget>
   <widget class="QMenu" name="menuOperations">
    <property name="title">
     <string>Operations</string>
    </property>
    <addaction name="actionInvertMsg"/>
    <addaction name="actionInvertKey"/>
   </widget>
   <addaction name="menuStyle"/>
   <addaction name="menuOperations"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionDark">
   <property name="text">
    <string>Dark</string>
   </property>
  </action>
  <action name="actionLight">
   <property name="text">
    <string>Light</string>
   </property>
  </action>
  <action name="actionInvertMsg">
   <property name="text">
    <string>Invert input R_i</string>
   </property>
  </action>
  <action name="actionInvertKey">
   <property name="text">
    <string>Invert key k_i</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QQuickWidget</class>
   <extends>QWidget</extends>
   <header location="global">QtQuickWidgets/QQuickWidget</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="feistel.qrc"/>
 </resources>
 <connections/>
</ui>
